clang-tools-extra/clangd/InlayHints.cpp

```diff
@@ -859,7 +872,9 @@ class InlayHintVisitor : public RecursiveASTVisitor<InlayHintVisitor> {
     else
       ForwardedParams = {Params.begin(), Params.end()};
 
-    NameVec ParameterNames = chooseParameterNames(ForwardedParams);
+    auto ForwardedParamsRef =
```

Comment:
What do you think about the following tweak:

```
SmallVector<const ParamVarDecl *> ForwardedParamsStorage;
ArrayRef<const ParamVarDecl *> ForwardedParams;
if (Callee.Decl) {
  ForwardedParamsStorage = resolveForwardingParameters(Callee.Decl);
  ForwardedParams = maybeDropCxxExplicitObjectParameters(ForwardedParamsStorage);
} else {
  ForwardedParams = Params;
}
```

Reply:
Thanks! The control flow looks much clearer now. BTW, I'd like to merge the calculation for `Param` above into this if block, too.